### PR TITLE
Add Automatic Git Hooks for Formatting, Clippy, and Build Checks

### LINKED ISSUE #289: Automatic lint checks using Git Hooks

### PR DESCRIPTION
**Pull request by Veercodeprog**
Related Issue:
Closes #289

### CODE REVIEW

File: ssr/src/fallback.rs [hunk 1]
```diff
@@ -26,7 +26,7 @@ pub async fn file_and_error_handler(
         handler(req).await.into_response()
     }
 }
-
+#[allow(unreachable_patterns)]
```
Comment (ravibazz) [inline ssr/src/fallback.rs:29]: The fix for this lint issue is already merged in main. Should we remove it?

**Review by ravibazz — CHANGES_REQUESTED** [review]
Hey @Veercodeprog Thank you for your contribution. Should we pull the latest changes from main to fix the lint issue?

File: ssr/src/fallback.rs (hunk 1 above)
Comment (Veercodeprog) [reply ssr/src/fallback.rs:29]: Thanks @ravibazz! Yes, let's pull the latest changes from main to resolve the lint issue and remove the redundant code.

Comment (Veercodeprog) [reply ssr/src/fallback.rs]: @ravibazz, I don't get any fallback errors in the main branch, but when I create a branch from main, I start getting this error. Can you explain why this is happening?

Comment (Veercodeprog) [reply ssr/src/fallback.rs]: Just checked the recent commit in main ,
I get it why this error occurs.

File: ssr/src/fallback.rs [hunk 2]
```diff
@@ -26,7 +26,7 @@ pub async fn file_and_error_handler(
         handler(req).await.into_response()
     }
 }
-
+// #[allow(unreachable_patterns)]
```
Comment (rupansh-gob) [inline ssr/src/fallback.rs:29]: lets remove this instead of commenting?

**Review by rupansh-gob — CHANGES_REQUESTED** [review]

File: ssr/src/fallback.rs (hunk 2 above)
Comment (Veercodeprog) [reply ssr/src/fallback.rs:29]: sure

Comment (Veercodeprog) [reply ssr/src/fallback.rs]: done @rupansh-gob

Comment (Veercodeprog) [reply ssr/src/fallback.rs]: opened a new pr for this , to resolve build failure.
should i close this one?

Comment (Veercodeprog) [reply ssr/src/fallback.rs]: New PR faces build failure as well—do we need to run or add a Fly script manually?

**Comment by rupansh-gob** [comment]
@Veercodeprog We have some ongoing CI changes so it was failing, I have approved your PR and it will be merged once the CI issues are sorted on our side

Thanks

**Review by ravibazz — APPROVED** [review]

**Comment by veer-asbytes** [comment]
adding new pr to fix build fail

**Pull request closed without merging**